system/modules/core/models/ArticleModel.php

```diff
@@ -203,6 +203,34 @@ public static function findPublishedByPidAndColumn($intPid, $strColumn, array $a
 
 
 	/**
+	 * Find all published articles with teaser by their parent ID
+	 *
+	 * @param integer $intPid     The page ID
+	 * @param array   $arrOptions An optional options array
+	 *
+	 * @return \Model\Collection|\ArticleModel[]|\ArticleModel|null A collection of models or null if there are no articles in the given column
+	 */
+	public static function findPublishedWithTeaserByPid($intPid, array $arrOptions=array())
+	{
+		$t = static::$strTable;
+		$arrColumns = array("$t.pid=? AND $t.showTeaser=1");
+
+		if (!BE_USER_LOGGED_IN)
```

Comment:
Checking for the backend user is a new and undesired case here, because the searchable pages must not be related to the user being logged in.